src/normalize/mod.rs

```diff
@@ -57,3 +61,50 @@ pub use user::UserOutput;
 
 pub(crate) use user::normalize_sorted_user_addrs_with_entries;
 pub(crate) use user::Handler;
+
+
+/// The reason why normalization failed.
+///
+/// The reason is generally only meant as a hint. Reasons reported may change
+/// over time and, hence, should not be relied upon for the correctness of the
+/// application.
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+#[non_exhaustive]
+pub enum Reason {
```

Comment:
Is it possible to detect and add 'permission denied'? On Android, it can happens that due to SELinux the caller can't access another process `/proc/<pid>/maps`. It would be good if we could surface that?

I guess you can do it in a follow-up diff anyway.

Reply:
As I said elsewhere, we already report permission denied errors as part of the result. It's just not bubbled up to the C API. We can rework that to return a more telling error.